src/utils/wallet.js

```diff
@@ -652,12 +652,18 @@ class Wallet {
         return new nearApiJs.Account(this.connection, accountId);
     }
 
-    async getAccount(accountId) {
+    async getAccount(accountId, limitedAccountData = false) {
         let account = new nearApiJs.Account(this.connection, accountId);
-        if (await TwoFactor.has2faEnabled(account)) {
+        const has2fa = await TwoFactor.has2faEnabled(account);
+        if (has2fa) {
             account = new TwoFactor(this, accountId);
         }
+        account.has2fa = has2fa;
```

Comment:
I was thinking about it when implementing this change, and I also think that it would be a much better implementation. But I'm a bit worried to dig into it right now because it looks like a bit more complicated change, and I would prefer to add an issue and work on it separately. The thing is that `has2faEnabled` is called every time you add a key, also when getting account data, also when recovering an account, and also on the `get2faMethod` method. I mean I would have to dig into each of these functionalities, to implement it.

I think the proper implementation would be to add `has2fa` to the constructor, and then only verify it if something about the 2fa has changed (like the user disabled it).

Are you ok with that? I mean working on it in a separate issue?